src/plugins/event-plugins/DomEventPlugin.ts

```diff
@@ -38,16 +38,22 @@ const defaultConfig: DomEventPluginConfig = {
     events: []
 };
 
+type ElementMap = {
+    element: HTMLElement;
+    eventListener: EventListener;
+};
+
 export class DomEventPlugin implements Plugin {
     private recordEvent: RecordEvent | undefined;
     private pluginId: string;
-    private eventListenerMap: Map<TargetDomEvent, EventListener>;
+    private eventListenerMap: Map<TargetDomEvent, ElementMap[]>;
```

Comment:
It allows us to keep track of the specific eventListener assigned to the element. Calling `this.getEventListener()` doesn't return the eventListener on the element but returns a new one so when it's time to remove the eventListener, it might be difficult to get the actual eventListener to remove.

To clarify, do you like adding the `element` to the `eventListenerMap` but not adding the array? But wouldn't we still need to keep track of the eventListener? Did you have another way in mind?

Reply:
As discussed, this change is beneficial in that (1) we can keep track of the event handlers to reference when removing the event handlers, and (2) we can keep track of the elements instead of searching for them in the DOM again. Please let me know if there are any other issues!

Reply:
Gotcha, thanks! Looks good.